packages/relay/src/TransactionManager.ts

```diff
@@ -183,24 +213,38 @@ data         | 0x${transaction.data.toString('hex')}
     return storedTx
   }
 
-  async resendTransaction (tx: StoredTransaction, currentBlock: number, newGasPrice: number, isMaxGasPriceReached: boolean): Promise<SignedTransactionDetails> {
+  async resendTransaction (tx: StoredTransaction, currentBlock: number, newMaxFee: number, newMaxPriorityFee: number, isMaxGasPriceReached: boolean): Promise<SignedTransactionDetails> {
     // Resend transaction with exactly the same values except for gas price
-    const txToSign = new Transaction(
-      {
-        to: tx.to,
-        gasLimit: tx.gas,
-        gasPrice: newGasPrice,
-        data: tx.data,
-        nonce: tx.nonce,
-        value: tx.value
-      },
-      this.rawTxOptions)
-
+    let txToSign: TypedTransaction
+    if (this.transactionType === TransactionType.TYPE_TWO) {
+      txToSign = new FeeMarketEIP1559Transaction(
+        {
+          to: tx.to,
+          gasLimit: tx.gas,
+          maxFeePerGas: newMaxFee,
+          maxPriorityFeePerGas: newMaxPriorityFee,
+          data: tx.data,
+          nonce: tx.nonce,
+          value: tx.value
+        },
+        this.rawTxOptions)
+    } else {
+      txToSign = new Transaction(
+        {
+          to: tx.to,
+          gasLimit: tx.gas,
+          gasPrice: newMaxFee,
```

Comment:
please note: there are places where you resolve to legacy with the priority, and others using the fee. 
its better to stick to one (even though they should be the same!) - I suggest to stick to fee, like here.

Reply:
I agree